lib/src/buffer/buffer.c

```diff
@@ -78,6 +78,43 @@ grow_buffer_if_needed(cardano_buffer_t* buffer, const size_t size_of_new_data)
   return CARDANO_SUCCESS;
 }
 
+/**
+ * \brief Deallocates a buffer object.
+ *
+ * This function is responsible for properly deallocating a buffer object (`cardano_buffer_t`)
+ * and its associated resources.
+ *
+ * \param object A void pointer to the buffer object to be deallocated. The function casts this
+ *               pointer to the appropriate type (`cardano_buffer_t*`).
+ *
+ * \note It is assumed that this function is called only when the reference count of the buffer
+ *       object reaches zero, as part of the reference counting mechanism implemented for managing the
+ *       lifecycle of these objects.
+ */
+static void
+cardano_buffer_deallocate(void* object)
+{
+  // Safe cast from void* to cardano_buffer_t*: This deallocator function is
+  // guaranteed to be called with a pointer to a `cardano_buffer_t` object. The use
+  // of a void pointer is necessary to maintain a generic interface for the
+  // deallocator callback, accommodating various object types.
+  // cppcheck-suppress misra-c2012-11.3
```

Comment:
cppcheck doesnt read these inline ignores. Lets supress the rule completely and add a deviation